src/node.zig

```diff
@@ -138,26 +139,19 @@ const Context = struct {
         assert(context.addresses.len > 0);
 
         const client_id = std.crypto.random.int(u128);
-
-        context.message_bus = try MessageBus.init(
-            allocator,
-            cluster,
-            context.addresses,
-            client_id,
-            context.io,
-        );
-        errdefer context.message_bus.deinit();
-
+        var message_pool = try MessagePool.init(allocator, .client);
```

Comment:
The client holds an address to the MessagePool, but the message pool is invalidated once this function returns.
Try storing the MessagePool inside the Context, making sure to also deinit() it later.